File: cha/bj/2559.js
const readline = require('readline');

const solve = (arr, n, k) => {
  // k일 동안 최대 온도
  let prev = 0;
  for (let i = 0; i < k; i += 1) {
    prev += arr[i];
  }
  let ans = prev;
  for (let i = k; i < n; i += 1) {
    prev -= arr[i - k];
    prev += arr[i];
    if (prev > ans) {
      ans = prev;
    }
  }
  return ans;
};

const input = () => {
  let n;
  let k;
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });
  rl.on('line', (line) => {
    if (!n) {
      [n, k] = line.split(' ').map((el) => +el);
    } else {
      const arr = line.split(' ').map((el) => +el);
      const ans = solve(arr, n, k);
      console.log(ans);
      rl.close();
    }
  }).on('close', () => process.exit());
};

input();

File: yang/20210329/2559.py
import sys
n,m=map(int,sys.stdin.readline().split())
k=list(map(int,sys.stdin.readline().split()))
a=0
maxa=0

for j in range(m):
    a+=k[j]

maxa=a

for i in range(len(k)-m):
    
    if m==1:
        print(max(k))
        exit()
    else:
        a-=k[i]
        a+=k[i+m]
        if a>maxa:
            maxa=a

print(maxa)